semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/CfgNodeMethods.scala

```diff
@@ -118,10 +119,14 @@ class CfgNodeMethods(val node: CfgNode) extends AnyVal with NodeExtension {
     node.lineNumber.map(_.toLong.toHexString)
   }
 
-  private def walkUpAst(node: CfgNode): Method =
+}
+
+object CfgNodeMethods {
+
+  def walkUpAst(node: CfgNode): Method =
```

Comment:
This should not be a public method, because it does not work for arbitrary cfg nodes. Only for some special few ones that are guaranteed to always be direct children of a method.

I was thinking we could change the parameter type to a union type - but that doesn't work because the `CallRepr` that's not a `Call` from line 107 only exists downstream in codescience, not here in joern. 

So I'd say let's just keep it private and duplicated.

Reply:
or actually, let's still move it to the object..